gcc/rust/ast/rust-ast.h

```diff
@@ -786,6 +773,41 @@ class DelimTokenTree : public TokenTree, public AttrInput
   {
     return AttrInput::AttrInputType::TOKEN_TREE;
   }
+
+  std::vector<std::unique_ptr<TokenTree> > &get_token_trees ()
+  {
+    return token_trees;
+  }
+
+  DelimType get_delim_type () const { return delim_type; }
+
+  static TokenId left_delim_type_tok_token_id (DelimType delim_type)
+  {
+    switch (delim_type)
+      {
+      case PARENS:
+	return LEFT_PAREN;
+      case SQUARE:
+	return LEFT_SQUARE;
+      case CURLY:
+	return LEFT_CURLY;
+      }
+    return LEFT_PAREN;
```

Comment:
I wonder if here it wouldn't be more "future-proof" to do an exhaustive switch statement and have a gcc_unreachable as the default case. This way, if the rust language decides to add new delimiters we'd find ourselves in a crash situation rather than a situation where the compiler tells us it's fine and assures us it's seen parentheses
```suggestion
    switch (delim_type)
      {
      case PARENS:
	return LEFT_PAREN;
      case SQUARE:
	return LEFT_SQUARE;
      case CURLY:
	return LEFT_CURLY;
      default:
        gcc_unreachable();
      }
```

Reply:
sounds good